src/controllers/krate/search.rs

```diff
@@ -32,15 +34,53 @@ use models::krate::{canon_crate_name, ALL_COLUMNS};
 /// function out to cover the different use cases, and create unit tests
 /// for them.
 pub fn search(req: &mut Request) -> CargoResult<Response> {
-    use diesel::sql_types::Bool;
-
     let conn = req.db_conn()?;
     let (offset, limit) = req.pagination(10, 100)?;
     let params = req.query();
-    let sort = params
+    let temp_params = params.clone();
+
+    let sort = temp_params
         .get("sort")
         .map(|s| &**s)
-        .unwrap_or("recent-downloads");
+        .unwrap_or("recent-downloads")
+        .clone();
+    let current_user_id = req.user()?.id;
+
+    let crates = execute_search(
+        &conn,
+        offset,
+        limit,
+        params,
+        sort.to_string(),
+        current_user_id,
+    );
+    let total = crates.len() as i64;
```

Comment:
This is not correct. The query is running the equivalent of `SELECT crates.*, recent_downloads.downloads, COUNT(*) OVER ()`. `total` is the number of crates that could be returned without pagination, not the number returned by the query.